modules/ROOT/pages/client-credentials.adoc

```diff
@@ -0,0 +1,297 @@
+    connection.close();
+  }
+
+  @Override
+  public ConnectionValidationResult validate(SalesforceClient connection) {
+    return success();
+  }
+}
+----
+
+The class implements ConnectionProvider, just like other connections in the SDK. Operations that are authenticated through the object operate over the SalesforceClient object that the provider returns. This is an important design consideration because it permits you to decouple the operation from the authentication method that is used because the connector could define another ConnectionProvider that uses  basic authentication, and all the operations remain fully compatible.
+
+=== @ClientCredentials
+
+This annotation indicates that this connection provider requires an OAuth dance using the Client Credentials grant type. The annotation has the following attributes:
+
+[source, java, linenums]
+----
+public @interface ClientCredentials {
+
+  /**
+   * @return The Url of the endpoint which provides the access tokens
+   */
+  String tokenUrl();
+
+  /**
+   * @return Expression to be used on the response of {@link #tokenUrl()} to extract the access token
+   */
+  String accessTokenExpr() default "#[payload.access_token]";
+
+  /**
+   * @return Expression to be used on the response of {@link #tokenUrl()} to extract the access token expiration
+   */
+  String expirationExpr() default "#[payload.expires_in]";
+
+  /**
+   * @return The default set of scopes to be requested, as a comma separated list. Empty string means no default scopes.
+   */
+  String defaultScopes() default "";
+
+  /**
+   * Allows to customize the placement that the client credentials will have in the request.
+   *
+   * @return the selected {@link CredentialsPlacement}. Defaults to {@link CredentialsPlacement#BASIC_AUTH_HEADER}
+   */
+  CredentialsPlacement credentialsPlacement() default BASIC_AUTH_HEADER;
+
+}
+----
+
+=== Connection Management Strategy
+
+The provider in the previous example does not implement any specialization of the ConnectionProvider interface, which means that the OAuth mechanism can be combined with the other connection management strategies. The connection objects can be pooled, cached, or created from scratch each time, depending on which interface is used (for example, PoolingConnectionProvider, CachedConnectionProvider, ConnectionProvider, and so on). For more information on connectivity management, see the connectivity reference.
+
+Be aware of the semantics of using the vanilla ConnectionProvider interface in this scenario. In a regular, “non-oauth” connection provider, using the vanilla interface means that each time a component requires a connection, a new one is created, and it is destroyed when the component is finished. Although this will remain true for the OAuth case, it does not mean that the OAuth dance is performed again. New connection objects are created, but the same access token is reused as long as it remains valid.
```

Comment:
"vanilla" is also idiomatic; consider "basic" or "regular" or "common" or a similar term that is appropriate to the specific context. GLOBAL.

Also, I think that the final two sentences can be merged with the one previous and minimized down to the points that a basic interface requires a new connection and later the destruction of that connection, but does not require a new access token.

Reply:
BREAK EDIT HERE

Reply:
```suggestion
Be aware of the semantics of using the non-customized ConnectionProvider interface in this scenario. In a regular, “non-OAuth” connection provider, using the non-customized interface means that each time a component requires a connection, a new one is created, and it is destroyed when the component is finished. Although this will remain true for the OAuth case, it does not mean that the OAuth dance is performed again. New connection objects are created, but the same access token is reused as long as it remains valid.
```

Reply:
Check word use. Vanilla cannot translate into Japanese properly.